### PR TITLE
Rename 'cases' to 'notifications' app wide- PSD-2147

### PR DESCRIPTION
**Pull request by acallaghan**
Renames 'Cases' to 'Notifications' app wide.

This is just for copy - a further sweep will rename code symbols and classes at a later date, as we now have three internal ways to describe notifications: `investigation`, `case`, & `notification`.

https://regulatorydelivery.atlassian.net/browse/PSD-2147

- Help pages and guides aren't changed - we already have a story to change these
- PRISM and support engines not changed - these will need separate tickets if needed

## Screen-shots or screen-capture of UI changes
CleanShot 2023-12-11 at 12 13 31
CleanShot 2023-12-11 at 12 13 41
CleanShot 2023-12-11 at 12 14 01
CleanShot 2023-12-11 at 12 14 19
CleanShot 2023-12-11 at 12 14 35
CleanShot 2023-12-11 at 12 17 07

## Review apps

https://psd-pr-2754.london.cloudapps.digital/

## Checklist:
- [x] Have you documented your changes in the pull request description?
- [x] Has acceptance criteria been tested by a peer?

### CODE REVIEW

File: app/controllers/bulk_products_controller.rb [hunk 1]
```diff
@@ -264,7 +264,7 @@
 
     if request.put?
       @bulk_products_upload.update!(submitted_at: Time.zone.now)
-      redirect_to all_products_path(sort_by: "created_at"), flash: { success: "The products were uploaded with the case number <a href=\"#{investigation_path(pretty_id: @bulk_products_upload.investigation.pretty_id)}\" class=\"govuk-link\">#{@bulk_products_upload.investigation.pretty_id}</a>".html_safe }
+      redirect_to all_products_path(sort_by: "created_at"), flash: { success: "The products were uploaded with the notification number <a href=\"#{investigation_path(pretty_id: @bulk_products_upload.investigation.pretty_id)}\" class=\"govuk-link\">#{@bulk_products_upload.investigation.pretty_id}</a>".html_safe }
```
[the remark this answers is not in the document]
Comment (acallaghan) [reply app/controllers/bulk_products_controller.rb:267]: the outputted link here isn't using user entered information, so isn't a risk

File: app/policies/investigation_policy.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ def readonly?
 
   # Used for all updating of the case, including adding and removing related
   # records, such as products, businesses and documents, with the exception of
-  # changing the case owner, its status (eg 'open' or 'closed'), and whether
+  # changing the otification owner, its status (eg 'open' or 'closed'), and whether
```
Comment (rubenarakelyan) [inline app/policies/investigation_policy.rb:9]: `notification` :)

Comment (acallaghan) [reply app/policies/investigation_policy.rb:9]: Cheers!

File: db/schema.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@
 
   create_table "active_storage_blobs", id: :serial, force: :cascade do |t|
     t.bigint "byte_size", null: false
-    t.string "checksum", null: false
```
Comment (rubenarakelyan) [inline db/schema.rb:43]: Should this be here?

**Review by rubenarakelyan — APPROVED** [review]

File: db/schema.rb (hunk 3 above)
Comment (acallaghan) [reply db/schema.rb:43]: Cheers, leak from Rails 7.1 upgrade

**Pull request merged**